src/components/GlobalNav/GlobalNav.jsx

```diff
@@ -1,7 +1,52 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import { Link } from 'react-router-dom';
+import './GlobalNav.scss';
 
-function GlobalNav(props) {
-  return <div>NavBar</div>;
+function GlobalNav() {
+  const [scrollPosition, setScrollPosition] = useState(false);
+
+  const checkscrollYPosition = () => {
+    (window.scrollY || document.documentElement.scrollTop) > 0
+      ? setScrollPosition(true)
+      : setScrollPosition(false);
+  };
```

Comment:
1. window.scrollY 와 document.documentElement.scrollTop이 같은 값이라고 생각되는데 둘 다 사용해야하는 이유가 있을까요?
2. 현재 로직에서도 결국 스크롤 값이 0보다 크면 매번 setScrollPosition 함수가 실행됩니다. 이럴 때는 scrollPosition의 값이 false이고 스크롤 값이 0보다 클때만 setScrollPosition(true)를 해준다 라는 개념으로 접근해보세요!

Reply:
안녕하세요 도현님! 늘 신경써주셔서 피드백 주시는 점 정말 감사합니다. 그밖의 사항은 모두 이해가 가서 수정완료 했는데, 해당 부분을 수정하다가 고민이 되는 부분이 있어서 의견 여쭤보고 싶습니다.
1. window.scrollY은 IE 최신버전에서 읽히지 않아서 IE 환경용 document.documentElement.scrollTop를 같이 사용하라는 글을 보고 참고했습니다. 
2. 만약 scrollposition의 값이 false일때,  scrollPosition이 0보다 크면 true로 변화를 주는 로직을 짜게 되면, 반대로 scrollPosition의 값이 true일때(이미 작아진 상태일때), scrollY깂이 다시 0이 되면 false로 돌아가는 함수를 또 설정해야 합니다. (스크롤이 꼭대기에 왔을 때 다시 넓어지는 점!)
3. 그렇다면 현재의 함수처럼 하나로 scrollPosition을 계속 관리하는 것과, 스크롤이 내려왔을때 키우는 함수/스크롤이 올라왔을 때 줄이는 함수 두 가지를 따로 써서 state를 관리하는 것 중 어느 쪽이 더 좋은 로직인가요?
감사합니다!

Reply:
1. 너무 좋습니다!
2. 아래와 같은 로직을 말씀드릴려고 했던겁니다. 아래 로직 참고해서 윤경님만의 코드로 조금 더 효율적으로 바꿔보세요~
```suggestion
  const checkscrollYPosition = () => {
    if(window.scrollY || document.documentElement.scrollTop < 0){
      setScrollPosition(false)
    } else {
      if(!scrollPosition) {
        setScrollPosition(true);
      }
    }
  };
```